File: CS/App_Code/ColumnUpdater.cs
using DevExpress.Web;
using System;
using System.Collections.Generic;
using System.Web.UI;

public class ColumnUpdater {
	List<GridViewColumn> columns;
	ASPxGridView _grid;
	Page page;
	ColumnUpdater(ASPxGridView currentGrid) {
		_grid = currentGrid;
		page = _grid.Page;
		if(page.IsPostBack)
			return;		
		_grid.DataBinding += ASPxGridView1_DataBinding;
		_grid.DataBound += ASPxGridView1_DataBound;
	}
	public static void Initialize(ASPxGridView currentGrid) {
		new ColumnUpdater(currentGrid);
	}
	void ASPxGridView1_DataBinding(object sender, EventArgs e) {		
		SaveColumns();
	}
	void ASPxGridView1_DataBound(object sender, EventArgs e) {
		UpdateColumns();
	}
	void SaveColumns() {
		columns = new List<GridViewColumn>();
		for(int i = 0; i < _grid.Columns.Count; i++) {
			columns.Add(_grid.Columns[i]);
		}
		_grid.Columns.Clear();
		_grid.AutoGenerateColumns = true;
	}
	void UpdateColumns() {
		foreach(GridViewColumn col in columns) {
			if(!(col is GridViewDataColumn)) {
				_grid.Columns.Add(col);
				continue;
			}
			for(int j = 0; j < _grid.Columns.Count; j++) {
				if(((GridViewDataColumn)col).FieldName != ((GridViewDataColumn)_grid.Columns[j]).FieldName)
					continue;
				_grid.Columns.RemoveAt(j);
				_grid.Columns.Add(col);
				break;
			}
		}
	}
}

File: VB/App_Code/ColumnUpdater.vb
Imports DevExpress.Web
Imports System
Imports System.Collections.Generic
Imports System.Web.UI

Public Class ColumnUpdater
    Private columns As List(Of GridViewColumn)
    Private _grid As ASPxGridView
    Private page As Page
    Private Sub New(ByVal currentGrid As ASPxGridView)
        _grid = currentGrid
        page = _grid.Page
        If page.IsPostBack Then
            Return
        End If
        AddHandler _grid.DataBinding, AddressOf ASPxGridView1_DataBinding
        AddHandler _grid.DataBound, AddressOf ASPxGridView1_DataBound
    End Sub
    Public Shared Sub Initialize(ByVal currentGrid As ASPxGridView)
        Dim TempColumnUpdater As ColumnUpdater = New ColumnUpdater(currentGrid)
    End Sub
    Private Sub ASPxGridView1_DataBinding(ByVal sender As Object, ByVal e As EventArgs)
        SaveColumns()
    End Sub
    Private Sub ASPxGridView1_DataBound(ByVal sender As Object, ByVal e As EventArgs)
        UpdateColumns()
    End Sub
    Private Sub SaveColumns()
        columns = New List(Of GridViewColumn)()
        For i As Integer = 0 To _grid.Columns.Count - 1
            columns.Add(_grid.Columns(i))
        Next i
        _grid.Columns.Clear()
        _grid.AutoGenerateColumns = True
    End Sub
    Private Sub UpdateColumns()
        For Each col As GridViewColumn In columns
            If Not(TypeOf col Is GridViewDataColumn) Then
                _grid.Columns.Add(col)
                Continue For
            End If
            Dim j As Integer = 0
            Do While j < _grid.Columns.Count
                If CType(col, GridViewDataColumn).FieldName <> CType(_grid.Columns(j), GridViewDataColumn).FieldName Then
                    j += 1
                    Continue Do
                End If
                _grid.Columns.RemoveAt(j)
                _grid.Columns.Add(col)
                Exit Do
                j += 1
            Loop
        Next col
    End Sub
End Class